packages/components/pie-button/src/index.ts

```diff
@@ -196,6 +197,23 @@ export class PieButton extends LitElement implements ButtonProps {
         this._handleClick();
     };
 
+    /**
+     * Template for the loading state
+     *
+     * @private
+     */
+    private renderSpinner (): TemplateResult {
+        const spinnerSize = this.size.includes('small') ? 's' : 'm';
+        const inverseVariants = ['primary', 'destructive', 'outline-inverse', 'ghost-inverse'];
```

Comment:
Shouldn't we rely on references instead of strings for this? Once a variant name changes (hopefully never 😆) this will become outdated.

Reply:
I might be missing your point, but these `strings` are a design contract only applicable to the spinner inside a button, if the button variant is one of the above, we need to use the inverse variant of the spinner and if ever needs updating, the logic is all located in the same code block. could you please elaborate more?

Reply:
My point is that strings are weak as references in the sense of which variants exist. 

`'primary', 'destructive', 'outline-inverse', 'ghost-inverse'` are a subset of the `variants` set in `defs.ts` correct?

The fact that the logic is in that block doesn't prevent us from making a mistake. If one of those variant names changes, let's say `outline-inverse`, since the `renderSpinner` has no connection with the `variants` from `defs.ts` it will not match the new name.

Reply:
could you please share what you think is the best approach here instead? is moving `inverseVariants` to `defs `would be enough?

Reply:
I'm not suggesting to use enums, but anything that relies on references would be good.

Reply:
@fernandofranca do you think that would fix the issue?

Reply:
i think the typing trick is neat and will ensure we don't use invalid values. 
updated 💯